### DISCUSSION TITLE
can't catch exception thrown in FutureProvider

### DISCUSSION THREAD

**Discussion by Yasser-saud** [rrousselGit/riverpod#1921] [Q&A] [answered]
Hello,

I can't seem to catch the exception thrown in the api provider when using FutureProvider.

i'm getting this every time

<img width="755" alt="Screen Shot 1444-04-25 at 2 41 49 AM" src="https://user-images.githubusercontent.com/60646608/202821004-2513ee08-fdef-4f15-bcee-db42e1ca61df.png">

small code snippet 

 `    
// just to test if i can catch this exception in the UI
      Future<List<Account>> getAccounts() async {
          await Future.delayed(const Duration(seconds: 2));
          throw Exception("Error");
       }

      final getAccountsProvider = FutureProvider.autoDispose<List<Account>>((ref) async {
        // this call will throw an exception
        return ref.read(apiProvider).getAccounts();
      });

                  // small snippet if the UI code
                  final accounts = ref.watch(getAccountsProvider);

                      itemExtent: 120,
                      children: accounts.when(
                        data: (data) {
                          return data
                              .map((e) => UserAvatar(
                                    id: e.accountId,
                                    avatar: e.avatar,
                                  ))
                              .toList();
                        },
                        error: (e, s) => [Text(e.toString())],
                        loading: () =>
                            [const CircularProgressIndicator.adaptive()],
                      ),
`

**Comment by rrousselGit** [comment, marked as answer]
The exception is caught by Riverpod. You will get `when(error:` displayed

Your IDE stops here because it's uncaught in your `FutureProvider`, that's all
